### ISSUE TITLE
Implement Windows-specific parts

### ISSUE THREAD

**Issue by edsko** [input-output-hk/ouroboros-network#285]
The storage layer has some Windows specific parts that need to be implemented and tested. 

Somewhat related: https://github.com/input-output-hk/ouroboros-network/pull/156 / https://github.com/input-output-hk/ouroboros-network/issues/170 .

**Comment by edsko** [comment]
Note on time estimate: Even though the actual work is not much, doing _anything_ on Windows is time consuming.

**Comment by edsko** [comment]
Related: If we implemented #292, the Windows bindings will need to be updated for that also.

**Comment by kderme** [comment]
I found an unfortunate conflict in the Win32 version we use. `katip` restricts Win32 >=2.3 && <2.5, however we will need Win32 >= 2.6.2.0, in order to have windows bindings for seek (known as setFilePointerEx  in the windows world) https://github.com/haskell/win32/commit/bcad524f56036662fa533f193098dd3036bdab00#diff-a99994770dbe04188c1176136aef3d32. 
```
$ cabal new-test test-storage --max-backjumps=20000 --extra-lib-dirs=/c/OpenSSL-Win64 --extra-include-dirs=/c/OpenSSL-Win64/include/
Resolving dependencies...
cabal.exe: Could not resolve dependencies:
[__0] trying: cardano-binary-1.5.0 (user goal)
[__1] trying: base-4.12.0.0/installed-4.1... (dependency of cardano-binary)
[__2] trying: aeson-1.4.3.0 (dependency of cardano-binary)
[__3] trying: cardano-crypto-wrapper-1.3.0 (user goal)
[__4] trying: base64-bytestring-type-1.0.1 (dependency of
cardano-crypto-wrapper)
[__5] trying: iohk-monitoring-0.1.4.0 (user goal)
[__6] next goal: katip (dependency of iohk-monitoring)
[__6] rejecting: katip-0.8.2.0, katip-0.8.1.0, katip-0.8.0.0 (conflict:
iohk-monitoring => katip<0.8)
[__6] trying: katip-0.7.0.0
[__7] next goal: Win32 (dependency of iohk-monitoring)
[__7] rejecting: Win32-2.6.1.0/installed-2.6..., Win32-2.8.3.0, Win32-2.8.2.0,
Win32-2.8.1.0, Win32-2.8.0.0, Win32-2.7.0.0, Win32-2.6.2.0, Win32-2.6.1.0,
Win32-2.6.0.0 (conflict: katip => Win32>=2.3 && <2.6)
[__7] trying: Win32-2.5.4.1
[__8] next goal: ouroboros-consensus (user goal)
[__8] rejecting: ouroboros-consensus-0.1.0.0 (conflict: Win32==2.5.4.1,
ouroboros-consensus => Win32==2.6.2.0)
[__8] fail (backjumping, conflict set: Win32, ouroboros-consensus)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: katip, Win32, aeson,
ouroboros-consensus, iohk-monitoring, base64-bytestring-type, base,
cardano-crypto-wrapper, cardano-binary
```
I'll need to think the less invasive way to fix it.

**Comment by mrBliss** [comment]
> I found an unfortunate conflict in the Win32 version we use. `katip` restricts Win32 >=2.3 && <2.6, however we will need Win32 >= 2.6.2.0, in order to have windows bindings for seek (known as setFilePointerEx in the windows world) [haskell/win32@bcad524#diff-a99994770dbe04188c1176136aef3d32](https://github.com/haskell/win32/commit/bcad524f56036662fa533f193098dd3036bdab00#diff-a99994770dbe04188c1176136aef3d32).
> 
> [..]
> 
> I'll need to think the less invasive way to fix it.

@kderme That's unfortunate. I currently see two options (feel free to suggest alternatives:
1. Make `katip` compatible with 2.6. Judging from the [changelog](https://github.com/haskell/win32/blob/master/changelog.md), I think https://github.com/haskell/win32/pull/83 is the breaking change that warranted the version bump. I haven't looked at the `katip` code, but making it work with 2.6 will probably be a lot of work, most of which is irrelevant to us.
2. Copy the`setFilePointerEx` implementation to some .hsc file under`src-win32`. It being a .hsc file might cause a Nix headache , though.

While 2 is not so clean, I think it is at least more time-efficient. What do you think?

**Comment by mrBliss** [comment]
@dcoutts We are assuming this does not need to be addressed for 1.0.0 because the exchanges are not using Windows, correct?

@kderme What's left to do for this ticket?

**Comment by kderme** [comment]
@mrBliss I think there's nothing left
